### PR TITLE
Add File Component and file handling for mechcomp signals

### PR DESCRIPTION
**Pull request by glassofmilk**
[ENHANCEMENT]

## About the PR 
This PR adds a data_file field onto mechanics signals, allowing the Wifi and PNet components to transmit any files attached to the packets they receive, as well as adding a new File Component to allow files to be saved and re-sent later. It also removes some code from the telepad that was preventing coordinates files from being received by other listeners.

## Why's this needed? 
Before this, you could sniff out other people's packets, but there was no way of intercepting their files. I think adding more interaction here is good, and it could incentivize development of other features that use files for communication. 

## Changelog

```
(u)glassofmilk:
(*)Added a new File Component to the MechComp Dispenser that allows saving and re-sending computer files
(+)Wifi and Powernet-networking Components now also transmit any files attached to packets they receive
```

### CODE REVIEW

**Comment by pali6** [comment]
@flrsh Also pending review.

File: code/WorkInProgress/MechanicMadness.dm [hunk 1]
```diff
@@ -1413,6 +1415,61 @@ var/list/mechanics_telepads = new/list()
...
+		mechanics.fireOutgoing(input)
+		animate_flash_color_fill(src,"#00FF00",2, 2)
+
+	proc/addandsendfile(var/datum/mechanicsMessage/input)
+		if (level == 2 || !src.stored_file) return
+		input.data_file = src.stored_file.copy_file()
+		mechanics.fireOutgoing(input)
+		animate_flash_color_fill(src,"#00FF00",2, 2)
+
+	proc/savefile(var/datum/mechanicsMessage/input)
```
Comment (ZeWaka) [inline code/WorkInProgress/MechanicMadness.dm:1453]: please rename this proc

File: code/WorkInProgress/MechanicsNetworkCon.dm [hunk 2]
```diff
@@ -106,29 +109,11 @@
 		var/dataStr = ""//list2params(S.data)  Using list2params() will result in weird glitches if the data already contains a set of params, like in terminal comms
 		for(var/i in S.data)
 			dataStr += "[i][isnull(S.data[i]) ? ";" : "=[S.data[i]];"]"
-		var/datum/mechanicsMessage/msg = mechanics.newSignal(dataStr)
+		var/datum/mechanicsMessage/msg = mechanics.newSignal(dataStr, S.data_file?.copy_file())
 		mechanics.fireOutgoing(msg)
 		animate_flash_color_fill(src,"#00AA00",1, 1)
 		return
 
-	proc/post_file(var/target_id, var/key, var/value, var/file)
```
Comment (ZeWaka) [inline code/WorkInProgress/MechanicsNetworkCon.dm:114]: You remove `post_file` here, but from what I can see, it's used by a bunch of procs in the codebase you haven't touched?

Comment (glassofmilk) [reply code/WorkInProgress/MechanicsNetworkCon.dm:114]: I don't think this is used anywhere. There's an /obj/machinery/networked/proc/post_file, but the Mechcomp network connector isn't overriding that - it's a new proc that isn't actually called by anything that I can see.

Comment (ZeWaka) [reply code/WorkInProgress/MechanicsNetworkCon.dm:114]: Ahh, a case of using the same name, got it.

**Pull request merged**